### PR TITLE
tests: lib: lte_lc_api: Added tests for some of the AT notifications

### PR DESCRIPTION
**Pull request by trantanen**
Added tests for some of the AT notifications, such as %NCELLMEAS (with and without gci), +CEREG, %XMODEMSLEEP and %MDMEV.

Supporting the notifications and a queue of them was added too for +CEREG notification.

Small logging improvement to lte_link_control library.

Jira: NCSDK-17957

### CODE REVIEW

**Comment by trantanen** [comment]
I'm still seeing if I have time to do more but I suspect not right now. Right now these test changes increases test coverage in `nrf/lib/lte_link_control` as follows:
```
Lines   (1523): 32.6% -> 61.9%
Functions (76): 57.9% -> 82.9%
Branches (863): 29.1% -> 50.8%
```

**Comment by trantanen** [comment]
@jtguggedal, @maxd-nordic, @simensrostad, please review this so we get it merged.

**Review by jtguggedal — APPROVED** [review]
Great to have these tests in, thanks for the effort!

File: tests/lib/lte_lc_api/src/lte_lc_api_test.c [hunk 1]
```diff
@@ -15,6 +15,189 @@
 
 #include "cmock_nrf_modem_at.h"
 
+#define TEST_EVENT_MAX_COUNT 5
+
+static struct lte_lc_evt test_event_data[TEST_EVENT_MAX_COUNT] = { 0 };
+static struct lte_lc_ncell test_neighbor_cells[CONFIG_LTE_NEIGHBOR_CELLS_MAX] = { 0 };
+static struct lte_lc_cell test_gci_cells[CONFIG_LTE_NEIGHBOR_CELLS_MAX] = { 0 };
```
[the remark this answers is not in the document]
Comment (trantanen) [reply tests/lib/lte_lc_api/src/lte_lc_api_test.c]: Fixed these. I don't know if compliance is somehow not applied for tests or why it doesn't complain.

File: tests/lib/lte_lc_api/prj.conf [hunk 2]
```diff
@@ -15,7 +16,8 @@ CONFIG_MOCK_NRF_MODEM_AT=y
 CONFIG_MOCK_NRF_MODEM_AT_SCANF_VARGS_COUNT=17
 
 # Enable logs if you want to explore them
-CONFIG_LOG=y
+#CONFIG_LOG=y
+#CONFIG_LTE_LINK_CONTROL_LOG_LEVEL_DBG=y
```
[the remark this answers is not in the document]
Comment (trantanen) [reply tests/lib/lte_lc_api/prj.conf:20]: I guess removing those is our guidance. And I disagree with it.
In general, this is a good way of documenting that these configs are something you might very likely want.
And in tests, log configurations are something you need immediately when you do development and see a failure in tests, which you cannot resolve based on test failure information.
If we don't have these here, people need to search and figure out what configs they need and that's not always straightforward.
I do understand that people shouldn't add too much uncommented configs into the .conf files but some are more helpful than distracting in my opinion.

[the remark this answers is not in the document]
Comment (jtguggedal) [reply tests/lib/lte_lc_api/prj.conf:20]: Yeah, I agree with your point and have used it myself for the same reasons. 
I'm fine with leaving them

**Review by simensrostad — APPROVED** [review]

**Comment by richardmccrae-nordicsemi** [comment]
I reran test-fw-nrfconnect-nrf_samples, which passed. Manual rebuilds of downstream jobs do not bubble back up to github I do not believe.

**Pull request merged**